visualiser/animation.py

```diff
@@ -0,0 +1,68 @@
+import copy
+
+import pydot
+import glob
+import os
+import shutil
+import imageio
+
+from visualiser import Node, Edge, Graph
+
+
+class Animation:
+    def __init__(self):
+        self.frames = []
+        self._frame_id = 0
+
+    def next_step(self, frame):
+        self.frames.append(copy.deepcopy(frame))
+        self._frame_id += 1
+
+    @staticmethod
+    def make_directory():
+        if not os.path.exists("frames"):
+            os.makedirs("frames")
+
+    def next_frame(self):
+        pass
+
+    def previous_frame(self):
+        pass
+
+    def get_frame(self, frame_id):
+        return self.frames[frame_id]
```

Comment:
**Suggestion:** Consider adding error handling for when the frame_id is not of type int. This will prevent the program from crashing when an invalid type is passed.
```suggestion
    def get_frame(self, frame_id):
            if not isinstance(frame_id, int):
                raise TypeError('frame_id must be of type int')
            return self.frames[frame_id]
```